src/openforms/contrib/kvk/migrations/0007_kvkconfig_profile_service_kvkconfig_search_service.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 4.2.10 on 2024-03-12 10:10
+
+from django.conf import settings
+from django.core.cache import caches
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def set_kvk_service(apps, _):
+    """
+    Derive the API root for KVK interaction.
+
+    This used to be configured in one Service. Due to possible gateways we have to go for
+    two separate Services since we need to provide different urls. The API root url can
+    be the same for both operations so in this case we use one Service configuration.
+    """
+    KVKConfig = apps.get_model("kvk", "KVKConfig")
+    config = KVKConfig.objects.first()
+    if config is None:
+        return
+
+    if service := config.service:
+        if not (root := service.api_root):
+            return
+
+    if "zoeken" in root:
+        config.search_service = service
+    elif "basisprofielen" in root:
+        config.profile_service = service
+    else:
+        # case where we have defined generic API root url (e.g. https://kvk.nl/test/api/)
+        config.search_service = service
+        config.profile_service = service
+
+    config.save()
+    caches[settings.SOLO_CACHE].clear()
```

Comment:
I'm not sure this is necessary. When working with models in migrations, we are dealing with "fake" models, and all the custom stuff added to model classes are not available. So I'm wondering if the solo cache exists at this point :thinking:

Reply:
Anyway it doesn't hurt to have this clear call :)

Reply:
if recently a request was made with KVK the value is populated in the cache, then the DB migration updates the config in the DB but the cache is still stale and will be re-used the next time the KVK feature is used.

So cache invalidation is required.